gh-90385: Add `pathlib.Path.walk()` method (#92517)
python/cpython: Misc/NEWS.d/next/Library/2022-05-10-16-30-40.gh-issue-90385.1_wBRQ.rst

> Added method walk to :class:`pathlib.Path` objects as a pathlib alternative for :func:`os.walk`

Reviewer: ```suggestion
pathlib.Path.walk() as an alternative for :func:`os.walk`.
```

Reviewer 2: Why partial markup?

```suggestion
Add :meth:`pathlib.Path.walk` as an alternative to :func:`os.walk`.
```
